test/webaudio/audioworklet.c

```diff
@@ -132,4 +143,9 @@ int main() {
   // and kick off Audio Worklet scope initialization, which shares the Wasm
   // Module and Memory to the AudioWorklet scope and initializes its stack.
   emscripten_start_wasm_audio_worklet_thread_async(context, wasmAudioWorkletStack, sizeof(wasmAudioWorkletStack), WebAudioWorkletThreadInitialized, 0);
+
+#ifdef TEST_AND_EXIT
+  // We're in the test harness and exiting is via main_thread_tls_access()
+  emscripten_exit_with_live_runtime();
+#endif
```

Comment:
I was thinking or a separate PR to enable audio testing CI.   

PR1: enable audio testing in CI
PR2: fix audio worklet test so they fail correctly

Reply:
Sure, NP. I’ll see how I get on with FF on Monday but I might split it for Chrome first too (I set up a VM with the same Debian and FF and so far, even with no audio device, it doesn’t fail; left to try is headless).

Reply:
After getting an audio-less Debian VM set-up without all the fallback sound devices removed, I finally managed to recreate what Firefox does. It fails here:

https://github.com/emscripten-core/emscripten/blob/f9a781b5b69161b31e3bdcd785fd7edadb24ddb8/test/webaudio/audioworklet.c#L60

The promise calls neither the success nor failure route, it just silently fails and the audio remains permanently suspended. I'll call this a Firefox bug and keep the `EMTEST_LACKS_SOUND_HARDWARE` (I'm satisfied that I've spent enough time going down all avenues and what remains is creating a repro in minimal JS for Firefox).